src/resources/js/tribe-events-bar.js

```diff
@@ -140,78 +142,166 @@ var tribe_events_bar_action;
 			$( document.getElementById( 'tribe-events-bar' ) ).addClass( 'tribe-has-datepicker' );
 		}
 
-		// Implement views links
-		function format( view ) {
-			return '<span class="tribe-icon-' + $.trim( view.text.toLowerCase() ) + '">' + view.text + '</span>';
-		}
-
 		// Implement placeholder
 		$( 'input[name*="tribe-bar-"]' ).placeholder();
 
 		// Create list
-		$( '<ul class="tribe-bar-views-list" />' ).insertAfter( $tribebarselect );
+		$( '<ul class="tribe-bar-views-list" tabindex="-1" role="listbox" aria-labelledby="tribe-bar-views-label" />' ).insertAfter( $tribebarselect );
 
 		var $tribebarviews = $( '.tribe-bar-views-list' );
 
+		function getCurrentViewItem() {
+			return $tribebarviews.find( 'li[data-view=' + $tribebarselect.find( ':selected' ).data( 'view' ) + ']' );
+		}
+
 		// Create list from select options
 		$tribebarselect.find( 'option' ).each( function( i ) {
 			var $view = $( this );
 			displaying = $view.data( 'view' );
 			// build list items and append them
 			var unique_c = 'tribe-bar-views-option-' + $view.data( 'view' );
 			$( '<li></li>', {
-				'class'               : 'tribe-bar-views-option ' + unique_c,
+				'id'                  : unique_c,
+				'class'               : 'tribe-bar-views-option',
```

Comment:
Are you removing `unique_c` from the classes for any reason in particular?

Reply:
Nothing more particular than it's redundant. I'm assigning that value as the `id` for each item now (since I need to reference the list items by ID for the `aria` elements to work properly). And, the actual icons displayed are handled via the icon `<span>` class, not the list item class.

I figured it was beneficial to remove unnecessary classes. Happy to add it back though if you feel otherwise.